app/objects/c_operation.py

```diff
@@ -328,33 +328,36 @@ async def _get_all_possible_abilities_by_agent(self, data_svc):
         return {a.paw: abilities for a in self.agents}
 
     def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors, agent_ran):
-        variables = re.findall(r'#{(.*?)}', self.decode_bytes(ability.test), flags=re.DOTALL) if ability.test else []
         if ability.ability_id in agent_ran:
             return
-        elif not agent.trusted:
+
+        ability_has_platform = any(executor for executor in ability.executors if executor.platform == agent.platform)
+        valid_executors = ability.get_executors(agent.platform, agent_executors)
```

Comment:
I'm wondering a few things:
1) Should the `ability_has_platform` check be moved to a method on `Ability`? 
2) Do we need to check `ability_has_platform` vs. the `valid_executors` check? That is, `valid_executors` will be empty if `ability_has_platform` is `False`.
3) If there is value in distinguishing no valid executors from not having the platform why don't we just return immediately when `ability_has_platform` is `False`?

Reply:
Removed the `ability_has_platform` check. The `valid_executors` check should cover all cases.

I think this makes more sense with the changes made in this PR.